akvo/iati/checks/fields/sectors.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+
+# Akvo RSR is covered by the GNU Affero General Public License.
+# See more details in the license.txt file located at the root folder of the Akvo RSR module.
+# For additional details on the GNU license please see < http://www.gnu.org/licenses/agpl.html >.
+
+from decimal import Decimal
+
+
+def sectors(project):
+    """
+    Check if sectors are present, either on project or transaction level, but not both.
+    If on transaction level, all transactions must have a sector.
+    Per sector vocabulary on project level, the percentages of the sectors should add up to 100%.
+    On transaction level, each sector vocabulary should only occur once per transaction.
+
+    :param project: Project object
+    :return: All checks passed boolean, [Check results]
+    """
+    checks = []
+    all_checks_passed = True
+
+    if project.sectors.all():
+        for transaction in project.transactions.all():
+            if transaction.sectors.all():
+                all_checks_passed = False
+                checks.append((u'error', u'sectors present both on project and transaction level'))
+
+        if all_checks_passed:
+            sectors_dict = {}
+            for sector in project.sectors.all():
+                try:
+                    sectors_dict[sector.vocabulary or '1'].append(sector.percentage)
+                except KeyError:
+                    sectors_dict[sector.vocabulary or '1'] = [sector.percentage]
+
+            for voc_key in sectors_dict.keys():
+                if len(sectors_dict[voc_key]) > 1:
+                    voc_percentage = 0
+
+                    for percentage in sectors_dict[voc_key]:
+                        try:
+                            voc_percentage += Decimal(percentage)
+                        except (ValueError, TypeError):
+                            all_checks_passed = False
+                            checks.append((u'error', u'multiple sectors with vocabulary \'%s\' '
```

Comment:
Python has two string delimiters for the very reason that you usually don't need to escape characters by using them :stuck_out_tongue_winking_eye: